test/src/edu/stanford/nlp/math/ADMathTest.java

```diff
@@ -0,0 +1,121 @@
+package edu.stanford.nlp.math;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class ADMathTest {
+
+  private void testCompareValues(DoubleAD correctResult, DoubleAD result, Double delta) {
+    assertEquals("Incorrect val", correctResult.getval(), result.getval(), delta);
+    assertEquals("Incorrect dot", correctResult.getdot(), result.getdot(), delta);
+  }
+
+  @Test
+  public void testMult() {
+    DoubleAD v1 = new DoubleAD(2.0, 4.0);
+    DoubleAD v2 = new DoubleAD(3.0, 2.0);
+
+    DoubleAD result = ADMath.mult(v1, v2);
+
+    DoubleAD correctResult = new DoubleAD(6.0, 16.0);
+    testCompareValues(correctResult, result, 0.0);
```

Comment:
There's nothing about this class that makes it use exact floating point calculations, so I believe 0.0 is not appropriate for the allowed delta.

Reply:
Thank you for the feedback the code. ` assertEquals("Incorrect dot", correctResult.getdot(), result.getdot(), delta); ` will be changed to ` correctResult.equals(result) ` and `correctResult.equals(result,TOL)`  following the equals implemented in the DoubleAD class. The change will occur in the next rebase. Thank you for the review, and sorry for late reply.